dvc/commands/experiments/ls.py

```diff
@@ -23,15 +27,20 @@ def run(self):
             git_remote=git_remote,
         )
 
+        if name_only or sha_only:
+            names = {}
+        elif isinstance(self.repo.scm, NoSCM):
+            raise SCMError("No SCM repository found")
```

Comment:
You don't need to check this here, we raise an exception as soon as `repo.experiments` is accessed if the repo is no-scm (so when we call `experiments.ls()`)